cpp/src/groupby/hash/aggregation_requests.cpp

```diff
@@ -76,21 +76,30 @@ struct avg_result_type {
   }
 };
 
-gdf_column* compute_average(gdf_column sum, gdf_column count,
-                            cudaStream_t stream) {
+gdf_column* compute_average(gdf_column sum, gdf_column count, cudaStream_t stream) {
   CUDF_EXPECTS(sum.size == count.size,
                "Size mismatch between sum and count columns.");
   gdf_column* avg = new gdf_column{};
+  gdf_binary_operator avg_binop = GDF_DIV;
 
   avg->dtype = cudf::type_dispatcher(sum.dtype, avg_result_type{});
+
+  // If the sum column is a GDF_TIMESTAMP, use floor_div binop instead of true_div
```

Comment:
What is this for?

Reply:
This is to support groupby mean aggregation on datetime columns. We can safely treat them like int64s everywhere else, but need to floor_div here because the return value shouldn't be a float.

Reply:
The result type is still going to be `GDF_FLOAT64` here, is that what you want?